### PR TITLE
Refactor the fork service slightly to take ForkRepoOptions

### PR DESCRIPTION
**Pull request by kevans91**
This reduces the number of places we need to change if we want to add other
options during fork time.

Signed-off-by: Kyle Evans <kevans@FreeBSD.org>

### CODE REVIEW

**Comment by kevans91** [comment]
I guess I should have mentioned: I have some local WIP that adds an option to disallow forking of a fork, the idea being that one could setup (in a highly controlled environment) something like: `org1/repo` forked to `user1/repo` with forking disabled on the latter, then forking is disabled on the former to prevent further forks.

**Comment by jolheiser** [comment]
> I guess I should have mentioned: I have some local WIP that adds an option to disallow forking of a fork, the idea being that one could setup (in a highly controlled environment) something like: `org1/repo` forked to `user1/repo` with forking disabled on the latter, then forking is disabled on the former to prevent further forks.

Just to clarify, there is still nothing inherently stopping someone from cloning and simply pushing to another remote.

**Comment by kevans91** [comment]
> > I guess I should have mentioned: I have some local WIP that adds an option to disallow forking of a fork, the idea being that one could setup (in a highly controlled environment) something like: `org1/repo` forked to `user1/repo` with forking disabled on the latter, then forking is disabled on the former to prevent further forks.
> 
> Just to clarify, there is still nothing inherently stopping someone from cloning and simply pushing to another remote.

Right, the point isn't to stop someone from cloning the repo, it's to stop them from taking up more space or providing an authoritative looking repository on "this" server.

File: models/repo.go [hunk 1]
```diff
@@ -1004,6 +1004,13 @@ type CreateRepoOptions struct {
 	MirrorInterval string
 }
 
+// ForkRepoOptions contains the fork repository options
+type ForkRepoOptions struct {
+	OldRepo     *Repository
```
Comment (lunny) [inline models/repo.go]: The name `OldRepo` is not right. Maybe `OriginalRepo` is a better name.

Comment (kevans91) [reply models/repo.go]: Sure, I had considered `OldRepo` or `ForkRepo` and stuck with the former to match the argument name, but IMO `OriginalRepo` (or maybe even `BaseRepo`?) is much more descriptive.

Comment (lunny) [reply models/repo.go]: I like personlly `OriginalRepo` or `SourceRepo`.

Comment (zeripath) [reply models/repo.go]: `models.Repo` has `BaseRepo` and `BaseRepoID` which is the id/repo of that was forked from.

I think therefore this should be `BaseRepo`.

Comment (kevans91) [reply models/repo.go]: I've switchedit to `BaseRepo` for consistency with the model.

**Review by zeripath — APPROVED** [review]

**Comment by zeripath** [comment]
Needs go fmt but otherwise looks good

**Comment by kevans91** [comment]
Bah, sorry, fixed.

**Review by lunny — APPROVED** [review]

**Pull request merged**